src/Controls/src/Core/Xaml/Diagnostics/BindablePropertyDiagnostics.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+
+namespace Microsoft.Maui.Controls.Xaml.Diagnostics;
+
+internal static class BindablePropertyDiagnostics
+{
+    public static ValueSource GetValueSource(BindableObject bindable, BindableProperty property)
+    {
+        if (bindable == null)
+            throw new ArgumentNullException(nameof(bindable));
+        if (property == null)
+            throw new ArgumentNullException(nameof(property));
+
+        var context = bindable.GetContext(property);
+        if (context == null)
+            return new ValueSource(BaseValueSource.Unknown);  
+
+        var specificity = context.Values.GetSpecificityAndValue().Key;
+        if (specificity == SetterSpecificity.DefaultValue)
+            return new ValueSource(BaseValueSource.Default);
+        if (specificity == SetterSpecificity.FromBinding)
+            return new ValueSource(BaseValueSource.Unknown, isExpression:true);
+        if (specificity == SetterSpecificity.ManualValueSetter)
+            return new ValueSource(BaseValueSource.Local);
+        if (specificity == SetterSpecificity.DynamicResourceSetter)
+            return new ValueSource(BaseValueSource.Unknown, isExpression:true);
+        if (specificity == SetterSpecificity.VisualStateSetter)
+            return new ValueSource(BaseValueSource.Style);
+        if (specificity == SetterSpecificity.Trigger)
+            return new ValueSource(BaseValueSource.StyleTrigger);
+        if (specificity == SetterSpecificity.FromHandler)
+            return new ValueSource(BaseValueSource.Unknown, isCurrent:true);
+        
+        if (specificity.Vsm > 0)
```

Comment:
@StephaneDelcroix need. some help fixing this merge.. I was trying something like this but not sure is correct. 

```
 if (specificity.IsVsm)
     return new ValueSource(BaseValueSource.Local);
	if (specificity.IsDefault)  
     return new ValueSource(BaseValueSource.Style);
 if (specificity.IsManual) 
     return new ValueSource(BaseValueSource.Local);
 if (specificity.IsDynamicResource)
     return new ValueSource(BaseValueSource.Unknown, isExpression:true);
 if (specificity.IsBinding)    
     return new ValueSource(BaseValueSource.Unknown, isExpression:true);
```